Include/unstable/objimpl.h

```diff
@@ -0,0 +1,111 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef _Py_UNSTABLE_OBJIMPL_H
+#  error "this header file must not be included directly"
```

Comment:
I was going to quibble about the directory name, but given this style of inclusion (i.e. only implicitly via the public header file when `Py_LIMITED_API` is *not* defined), I think 'unstable' is OK.